### ISSUE TITLE
Expand functionality on knowing about retractions

### ISSUE THREAD

**Issue by svenvanderburg** [fair-workflows/nanopub#83]
It is clear from the documentation how you can retract a nanopublication, but it is unclear how you can know whether a given nanopublication is retracted.

Requirements:
* [ ] Add a Publication.is_retracted property to the Publication class, it should do a get request using NanopubClient.find_nanopubs_with_pattern to see whether the publication is retracted.
* [x] Add to documentation (with a bit of context)
* [x] Add a `find_retractions_of` method that returns a list of publications that retract a given nanopub

**Comment by svenvanderburg** [comment]
We could ideally also add a warning if users fetch a nanopub for which there exists a retraction nanopub.

**Comment by vemonet** [comment]
The problem is deeper, most Nanopubs call (in the python lib, and in the grlc API) do not even filter out retracted publications, preventing to get the not retracted nanopubs

You can easily filter out retracted nanopubs this way though:

```sparql
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX bl: <https://w3id.org/biolink/vocab/>
PREFIX np: <http://www.nanopub.org/nschema#>
PREFIX npx: <http://purl.org/nanopub/x/>
SELECT * WHERE {
  graph ?np_assertion {
    ?association
      bl:association_type ?association_type ;
      rdf:subject ?subject ;
      rdf:predicate ?predicate ;
      rdf:object ?object ;
      bl:relation ?relation ;
      bl:provided_by ?provided_by .
  }

  graph ?np_head {
    ?np_uri np:hasAssertion ?np_assertion .
  }
  FILTER NOT EXISTS { ?creator npx:retracts ?np_uri }
}
```

For more accuracy you can potentially also map the `?creator` to make sure the person retracting is the same publishing

**Comment by raar1** [comment]
Thanks a lot @vemonet ! If I understand correctly, mapping the ```?creator``` would also involve a check that the RSA key they signed the retraction with is the same as that used to sign the original claim?

**Comment by vemonet** [comment]
Theorically, but I did not tried it

I posted more about using `find_nanopubs_with_pattern` to delete Nanopubs :

https://github.com/fair-workflows/nanopub/issues/84

One weird thing is that I needed to force the deletion because according to the lib it was not the same pubkey, but I published those nanopubs yesterday using the python lib and the same keys

I did not digged this though :p

**Comment by raar1** [comment]
Ahhh ok, yes, we added the retraction functionality very recently, so I am not surprised at these issues springing up. Thanks for testing. I will reply on #84 and check with Tobias if we could update the GRLC API.

**Comment by raar1** [comment]
> One weird thing is that I needed to force the deletion because according to the lib it was not the same pubkey, but I published those nanopubs yesterday using the python lib and the same keys

If you have time would you mind opening an issue for that as well? You can just dump your code and maybe the start and end of the public key line in .nanopub/. I think the key check is fairly naive so I wouldn't be surprised if it's failing because there is a host name given in the line or something like that. If so, we definitely need to fix that.

**Comment by vemonet** [comment]
@raar1 I tried again without the `force=true` and now it works without problem, I will post next time I face it

**Comment by svenvanderburg** [comment]
Fixed in #93 
I did not add a is_retracted property to the Publication class, because we would have to instantiate a NanopubClient in the Publication object. Then you get all kinds of circular import problems, and the problem that you cannot specify whether you want to use test or prod server. So users just have to use NanopubClient.find_retractions_of to know whether a publication was retracted (I added documentation on this).
